DataFormats/Candidate/interface/LeafCandidate.h

```diff
@@ -33,6 +33,8 @@ namespace reco {
 
     typedef unsigned int index;
 
+    LeafCandidate() {}
```

Comment:
Technically yes, but the result will be the same.  If a LeafCandidate is constructed with no arguments, the default ctor will be used instead of the variadic templated one, but in either case all of the data members will be default constructed. I will have a local C++ guru look at this shortly to verify that this is correct.